### DISCUSSION TITLE
Gestion des applications en cluster

### DISCUSSION THREAD

**Discussion by VinceBlood** [dbarzin/mercator#122] [Q&A]
Bonjour,

Nos systèmes sont doublés a tout niveau, cad matériel (serveur1,2) et logiciel (service main, backup).

Quel est la bonne pratique pour la partie logicielle ?

Doit-on tout doubler ?

Exemple :
Application : Base de données "Macro-Application"
Services : Instance1 -> Serveur1, Instance2 -> Serveur 2, Witness -> Serveur 3
Module : (Instance1):SQLServer-1433, (Instance1):SQLAgent-JobScheduler

**Comment by dbarzin** [comment]
Il n'est pas nécessaire de tout doubler, cela dépend du type de haute disponibilité que vous avez mis en place.
Une application peut être déployée sur plusieurs serveurs virtuels qui sont eux même déployés sur un ou plusieurs serveurs physiques.

**Comment by VinceBlood** [comment]
Bonjour,

Il s'agit dans la plupart des cas, de cluster applicatif miroir. C'est à dire qu'il y a une application active et autre standby.
Le cas de SQL server est different, c'est du Actif/Actif avec une notion de VIP, ca voudrait dire qu'il faut déclarer un serveur logique VIP.

**Reply by dbarzin** [reply]
Oui, si vous avez deux serveurs logiques différents, il faut déclarer deux serveurs dans la cartographie.
Notez qu'en mode miroir, il ne faut déclarer qu'un serveur logique qui est déployé sur plusieurs serveur physiques.

**Comment by VinceBlood** [comment]
Merci,

Je vois le principe, ca marchera dans des configurations où j'imagine que la VM se déplace d'un noeud à l'autre.

Nous dupliquons également les VMs.

Ca donnerait donc :

Exemple Cluster VMWare :
Application -> ServeurLogique -> ESX1, ESX2

Exemple VMWare Standalone :
Application -> ServeurLogique1 -> ESX1
Application -> ServeurLogique2 -> ESX2

Et lorsque nous avons des environnements différents (prod, preprod, formation, test).
Nous avons aussi ces applications.
L'idée est de conserver l'application plutôt que d'en faire des variantes ?
(Les versions sont differentes selon l'environnement)

Ca pourrait etre bien pour chaque application de rappeler les dependances dans une colonne.

Par exemple

Application | Dependance
Base données | ServeurLogique.ESX1
               ServeurLogique.ESX2

**Comment by dbarzin** [comment]
La notion d'environnement (prod, preprod, test, dev...) est lié au serveur logique sur lequel se trouve l'application.
On essaye de ne pas dupliquer les applications mais vous pouvez toujours le faire en ajoutant par exemple ".dev" au nom de l'application.
Il est possible de retrouver les dépendances dans le rapport "liste des applications par groupe applicatif".

**Comment by VinceBlood** [comment]
Ah je vois, il faut donc voir l'application (+service) comme une reference à un catalogue d'application. L'application réelle étant l'association d'une application de catalogue (et ses services associés) à un serveur logique.

**Comment by dbarzin** [comment]
Oui, c'est bien exprimé comme cela.
